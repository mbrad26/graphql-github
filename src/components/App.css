#url {
  width: 300px;
}
