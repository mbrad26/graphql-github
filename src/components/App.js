import React, { useState, useEffect, useCallback } from 'react';

import './App.css'
import Form from './Form';
import Organization from './Organization';
import { axiosGitHubGraphQL } from '../api/api';
import {
  GET_ISSUES_OF_REPOSITORY,
 } from '../Queries/queries';

const TITLE = 'React GraphQL GitHub Client';

const App = () => {

  console.log('APP');

  const [organization, setOrganization] = useState();
  const [errors, setErrors] = useState();

  const loadingData = useCallback(() => {
    axiosGitHubGraphQL
      .post('', { query: GET_ISSUES_OF_REPOSITORY })
      .then(res => {
        console.log(res);
        setOrganization(res.data.data.organization);
        setErrors(res.data.errors);
      });
  }, []);

  useEffect(() => {
    loadingData()
  }, [loadingData]);

  return (
    <div>
      <h1>{TITLE}</h1>
      <Form />
      <hr />
      {organization
        ? <Organization organization={organization} errors={errors}/>
        : <p>Loading data ...</p>
      }
    </div>
  );
};


export default App;
