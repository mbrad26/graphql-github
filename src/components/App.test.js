import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App.js';

describe('App', () => {
  it('renders snapshot', () =>{
    render(<App />);

    screen.debug();

    expect(screen.getByText('React GraphQL GitHub Client')).toBeInTheDocument();
  });
});
