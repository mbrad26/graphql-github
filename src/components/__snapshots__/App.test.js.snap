// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<div>
  <h1>
    React GraphQL GitHub Client
  </h1>
  <form>
    <label
      for="url"
    >
      Show open issues for https://github.com/
    </label>
    <input
      id="url"
      type="text"
      value="the-road-to-learn-react/the-road-to-learn-react"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
  <hr />
  <p>
    Loading data ...
  </p>
</div>
`;
